receiver/singletonreceivercreator/factory.go

```diff
@@ -35,8 +36,26 @@ func createDefaultConfig() component.Config {
 	}
 }
 
-func createMetricsReceiver(_ context.Context, params receiver.Settings, cfg component.Config, consumer consumer.Metrics) (receiver.Metrics, error) {
-	r := newSingletonReceiverCreator(params, cfg.(*Config))
-	r.nextMetricsConsumer = consumer
-	return r, nil
+func createMetricsReceiver(
```

Comment:
the signature format is different from rest of the repo

Reply:
As discussed, this format is pretty common (we are even using it in telemetry-manager) if the argument list gets too long, which is often the case for ctor functions (`newXxx`)